src/Hosting/Hosting/src/WebHostBuilderExtensions.cs

```diff
@@ -67,7 +71,7 @@ private static IWebHostBuilder Configure(this IWebHostBuilder hostBuilder, Actio
         /// <param name="hostBuilder">The <see cref="IWebHostBuilder"/> to configure.</param>
         /// <param name="startupFactory">A delegate that specifies a factory for the startup class.</param>
         /// <returns>The <see cref="IWebHostBuilder"/>.</returns>
-        public static IWebHostBuilder UseStartup(this IWebHostBuilder hostBuilder, Func<WebHostBuilderContext, object> startupFactory)
+        public static IWebHostBuilder UseStartup<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)]TStartup>(this IWebHostBuilder hostBuilder, Func<WebHostBuilderContext, TStartup> startupFactory)
```

Comment:
Huh, I didn't know you use attributes like this. Is it new?

Reply:
Yes https://github.com/dotnet/designs/blob/master/accepted/2020/linking-libraries.md

Reply:
Thank god this is new, I saw it on twitter and I was starting to feel dumb after 10+ years not knowing you could do this :)